(WIP) New HPO features implemented

eunwoosh opened the pull request:
This PR contains new HPO features.
Each features are as bellow

- hyperband is enabled (single GPU ASHA)
- model weight trained during HPO is used by resume
- initial weight of each trials is fixed for stable HPO

eunwoosh committed:
hpo test

eunwoosh committed:
pass weight after hpo

eunwoosh committed:
minor change

eunwoosh committed:
resume bugfix

eunwoosh committed:
eval mAP score

eunwoosh committed:
pass default hyper parameters to hpopt and set mimimum iter of asha depending on warm iter

eunwoosh committed:
apply resume to cls and seg also

eunwoosh committed:
bugfix for hpo resume wiht cls

eunwoosh committed:
update hyperband, resume and fixed initial weight

eunwoosh committed:
code for copying weight is moved into deep-object-reid

yunchu commented on ote_cli/ote_cli/utils/hpo.py:
```diff
@@ -298,20 +366,42 @@ def __init__(
             non_pure_train_ratio=val_dataset_size
             / (train_dataset_size + val_dataset_size),
             batch_size_name="learning_parameters.batch_size",
+            default_hyper_parameters=default_hyper_parameters
```
can we add things below at the bottom of this line?
`            metric=self.metric,
            mode=hpopt_cfg.get("mode", "max")
`
I've found that such setting in the hp_config.yaml is not passed to the hpopt.

yunchu commented on ote_cli/ote_cli/utils/hpo.py:
```diff
@@ -457,7 +552,28 @@ def run(self):
+
+            for file_name in os.listdir(hpo_weight_path):
+                if "best" in file_name:
+                    hpo_weight_path += f"/{file_name}"
```
use os.path.join() to merge file path

yunchu commented on external/mmsegmentation/segmentation_tasks/apis/segmentation/inference_task.py:
```diff
@@ -116,6 +116,10 @@ def _load_model(self, model: ModelEntity):
+
+                if "load_from" in self._config:
+                    self._config.load_from = None
+
```
what is the purpose of this?

eunwoosh committed:
pass additional parameter to hpopt

eunwoosh committed:
minor fix and load_from with hpo fix

yunchu committed:
updated hpo

to adjust asha internal by applying changes made by adaptive-repeat
of the mmdet and mmseg

Signed-off-by: Lee, Yunchu <yunchu.lee@intel.com>
